ACM-37751 - NetworkPolicy md (#649)
stolostron/insights-client: docs/NETWORK_POLICIES.md

  `kubernetes.default.svc` ClusterIP traffic through the OVN service load balancer *before*
  NetworkPolicy evaluation, so no egress rule type can match kube-API traffic. Applying an
  Egress policyType would silently block insights-client from reaching the Kubernetes API.
  insights-client also needs egress to the external CCX/Insights endpoint
  (`console.redhat.com`), which further rules out deny-all egress.
- **Well-known namespace labels.** Ingress rules that reference OpenShift system namespaces
  use the `kubernetes.io/metadata.name` label, which the API server automatically stamps on
  every namespace (Kubernetes 1.21+). This avoids relying on custom labels that may not exist
  in every cluster.

## Component network flows

### insights-client

| Direction | Peer | Port | Rationale |
|---|---|---|---|
| Ingress | *(none allowed)* | — | insights-client does not serve any active endpoints. It runs an HTTP/TLS listener on port 3030 but has no registered route handlers, so no external traffic needs to reach it. The policy selects the pod and declares `Ingress` policyType with no ingress rules, effectively denying all inbound traffic. |

> | Egress | *(not restricted — Ingress-only policy)* | — | insights-client requires egress to the Kubernetes API (watch ManagedClusters, read pull-secrets, CRUD PolicyReports, read ClusterVersions and APIServer config) and to the external CCX/Insights endpoint (`console.redhat.com`, port 443) to fetch cluster reports. OVN-Kubernetes handles `kubernetes.default.svc` ClusterIP traffic via the OVN service load balancer before NetworkPolicy evaluation, so no egress rule can match kube-API traffic. Applying an Egress policyType would silently block both API server access and external report retrieval. |

Reviewer: 🤖 I'm an AI agent

Same minor wording suggestion as in the sister PR stolostron/insights-metrics#529 — "not restricted by this NetworkPolicy" is more precise than "not restricted — Ingress-only policy", since other NetworkPolicies selecting the same pods could still restrict egress.

```suggestion
| Egress | *(not restricted by this NetworkPolicy)* | — | insights-client requires egress to the Kubernetes API (watch ManagedClusters, read pull-secrets, CRUD PolicyReports, read ClusterVersions and APIServer config) and to the external CCX/Insights endpoint (`console.redhat.com`, port 443) to fetch cluster reports. OVN-Kubernetes handles `kubernetes.default.svc` ClusterIP traffic via the OVN service load balancer before NetworkPolicy evaluation, so no egress rule can match kube-API traffic. Applying an Egress policyType would silently block both API server access and external report retrieval. |
```

Author: applied
